notificationworkerlambda/src/main/scala/com/gu/notifications/worker/tokens/SqsDeliveryService.scala

```diff
@@ -0,0 +1,66 @@
+package com.gu.notifications.worker.tokens
+
+import java.util.concurrent.TimeUnit
+
+import cats.effect._
+import com.amazonaws.handlers.AsyncHandler
+import com.amazonaws.regions.Regions
+import com.amazonaws.services.sqs.AmazonSQSAsyncClientBuilder
+import com.amazonaws.services.sqs.model.{SendMessageRequest, SendMessageResult}
+import com.gu.notifications.worker.utils.Aws
+import fs2.Stream
+import play.api.libs.json.Json
+
+import scala.concurrent.ExecutionContextExecutor
+import scala.concurrent.duration.FiniteDuration
+import scala.util.Random
+
+trait SqsDeliveryService[F[_]]{
+  def sending(chunkedTokens: ChunkedTokens): Stream[F, Either[Throwable, Unit]]
+}
+class SqsDeliveryServiceImpl[F[_]](queueUrl: String)(implicit ece: ExecutionContextExecutor,
+  contextShift: Concurrent[F],
+  F: Async[F],
+  T: Timer[F]
+) extends SqsDeliveryService[F]{
+  val amazonSQSAsyncClient = AmazonSQSAsyncClientBuilder.standard()
+    .withCredentials(Aws.credentialsProvider)
+    .withRegion(Regions.EU_WEST_1)
+    .build()
+
+
+  private def send(chunkedTokens: ChunkedTokens)(oncomplete: Either[Exception, Unit] => Unit): Unit = {
+    amazonSQSAsyncClient.sendMessageAsync(new SendMessageRequest()
+      .withMessageBody(Json.stringify(Json.toJson(chunkedTokens)))
+        .withQueueUrl(queueUrl)
+      , new AsyncHandler[SendMessageRequest, SendMessageResult] {
```

Comment:
Can you use AWSAsync.wrapAsyncMethod here?

Reply:
That returns a Future and I wasn't sure whether Future's were best to mix in with fs2. Let's chat about this.

Reply:
Oh yeah I just realised it was handled with a callback. Fine :+1: